src/Core/Authentication.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * @file /src/Core/Authentication.php
+ */
+
+namespace Friendica\Core;
+
+use Friendica\Core\Addon;
+use Friendica\BaseObject;
+use Friendica\Core\Config;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Database\DBA;
+use Friendica\Core\L10n;
+use Friendica\Core\PConfig;
+
+/**
+* Handle Authentification, Session and Cookies
+*/    
+class Authentication extends BaseObject
+{
+	/**
+	 * @brief Calculate the hash that is needed for the "Friendica" cookie
+	 *
+	 * @param array $user Record from "user" table
+	 *
+	 * @return string Hashed data
+	 */
+	public static function cookie_hash($user)
+	{
+		return(hash("sha256", Config::get("system", "site_prvkey") .
+				$user["prvkey"] .
+				$user["password"]));
+	}
+
+	/**
+	 * @brief Set the "Friendica" cookie
+	 *
+	 * @param int $time
+	 * @param array $user Record from "user" table
+	 */
+	public static  function new_cookie($time, $user = [])
+	{
+		if ($time != 0) {
+			$time = $time + time();
+		}
+
+		if ($user) {
+			$value = json_encode(["uid" => $user["uid"],
+				"hash" => self::cookie_hash($user),
+				"ip" => defaults($_SERVER, 'REMOTE_ADDR', '0.0.0.0')]);
+		} else {
+			$value = "";
+		}
+
+		setcookie("Friendica", $value, $time, "/", "", (Config::get('system', 'ssl_policy') == SSL_POLICY_FULL), true);
+	}
+
+	/**
+	 * @brief Sets the provided user's authenticated session
+	 *
+	 * @todo Should be moved to Friendica\Core\Session once it's created
+	 *
+	 * @param type $user_record
+	 * @param type $login_initial
+	 * @param type $interactive
+	 * @param type $login_refresh
+	 */
+	public static function success($user_record, $login_initial = false, $interactive = false, $login_refresh = false)
+	{
+		$a = self::getApp();
+
+		$_SESSION['uid'] = $user_record['uid'];
+		$_SESSION['theme'] = $user_record['theme'];
+		$_SESSION['mobile-theme'] = PConfig::get($user_record['uid'], 'system', 'mobile_theme');
+		$_SESSION['authenticated'] = 1;
+		$_SESSION['page_flags'] = $user_record['page-flags'];
+		$_SESSION['my_url'] = $a->getbaseUrl() . '/profile/' . $user_record['nickname'];
+		$_SESSION['my_address'] = $user_record['nickname'] . '@' . substr($a->getbaseUrl(), strpos($a->getbaseUrl(), '://') + 3);
+		$_SESSION['addr'] = defaults($_SERVER, 'REMOTE_ADDR', '0.0.0.0');
+
+		$a->user = $user_record;
+
+		if ($interactive) {
+			if ($a->user['login_date'] <= NULL_DATE) {
+				$_SESSION['return_url'] = 'profile_photo/new';
+				$a->module = 'profile_photo';
+				info(L10n::t("Welcome ") . $a->user['username'] . EOL);
```

Comment:
Watch out for global functions, their corresponding file should be `require_once` by this file until we move them to classes.